[BUG] Compilation error with NO_MOTION_BEFORE_HOMING & HOME_AFTER_DEACTIVATE

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

Marlin\src\gcode\motion\../../module/motion.h:584:53: error: 'homing_needed_error' was not declared in this scope  584 |   #define MOTION_CONDITIONS (marlin.isRunning() && !homing_needed_error())
      |                                                     ^~~~~~~~~~~~~~~~~~~

### Bug Timeline

_No response_

### Expected behavior

_No response_

### Actual behavior

_No response_

### Steps to Reproduce

_No response_

### Version of Marlin Firmware

2026-02-11

### Printer model

any

### Electronics

_No response_

### LCD/Controller

_No response_

### Other add-ons

_No response_

### Bed Leveling

None

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [x] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[marlin.zip](https://github.com/user-attachments/files/25296593/marlin.zip)

--- a/Marlin/src/module/motion.h
+++ b/Marlin/src/module/motion.h
@@ -581,7 +581,7 @@ inline float home_bump_mm(const AxisEnum axis) {
 #endif
 
 #if ENABLED(NO_MOTION_BEFORE_HOMING)
-  #define MOTION_CONDITIONS (marlin.isRunning() && !homing_needed_error())
+  #define MOTION_CONDITIONS (marlin.isRunning() && !motion.homing_needed_error())
 #else
   #define MOTION_CONDITIONS marlin.isRunning()
 #endif

